hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlockIndex.java

```diff
@@ -164,13 +158,14 @@ public boolean isEmpty() {
     @Override
     public BlockWithScanInfo loadDataBlockWithScanInfo(Cell key, HFileBlock currentBlock,
         boolean cacheBlocks, boolean pread, boolean isCompaction,
-        DataBlockEncoding expectedDataBlockEncoding) throws IOException {
+        DataBlockEncoding expectedDataBlockEncoding,
+        CachingBlockReader cachingBlockReader) throws IOException {
```

Comment:
What is the new parameter CachingBlockReader used for?

Reply:
CellBasedKeyBlockIndexReader will use this, add it as parameter in order to make BlockIndexReader stateless, so we can share it between readers.